integration/integrationtest/docker.go

```diff
@@ -109,9 +103,15 @@ func RunEnvbox(t *testing.T, pool *dockertest.Pool, conf *CreateDockerCVMConfig)
 		host.ExtraHosts = []string{"host.docker.internal:host-gateway"}
 	})
 	require.NoError(t, err)
-	// t.Cleanup(func() { _ = pool.Purge(resource) })
 
-	waitForCVM(t, pool, resource)
+	t.Cleanup(func() {
+		if !t.Failed() {
+			_ = pool.Purge(resource)
+		}
+	})
+
+	success := waitForCVM(t, pool, resource)
+	require.Equal(t, success, !conf.ExpectFailure, "expected success=%v but detected %v", !conf.ExpectFailure, success)
```

Comment:
Expected comes first!

```suggestion
	require.Equal(t, !conf.ExpectFailure, success, "expected success=%v but detected %v", !conf.ExpectFailure, success)
```